Sources/OktaDirectAuth/Internal/Step Handlers/OOBStepHandler.swift

```diff
@@ -0,0 +1,164 @@
+                                           mfaToken: self.mfaToken,
+                                           oobCode: response.oobCode,
+                                           grantTypesSupported: self.flow.supportedGrantTypes)
+                self.poll = PollingHandler(client: self.flow.client,
+                                           request: request,
+                                           expiresIn: response.expiresIn,
+                                           interval: response.interval) { pollHandler, result in
+                    switch result {
+                    case .success(let response):
+                        return .success(response.result)
+                    case .failure(let error):
+                        guard case let .serverError(serverError) = error,
+                           let oauthError = serverError as? OAuth2ServerError
+                        else {
+                            return .failure(error)
+                        }
+
+                        switch oauthError.code {
+                        case .slowDown:
+                            pollHandler.interval += 5
+                            fallthrough
+                        case .authorizationPending: fallthrough
+                        case .directAuthAuthorizationPending:
+                            return .continuePolling
+                        default:
+                            return .failure(error)
+                        }
+                    }
+                }
+                
+                self.poll?.start { result in
+                    switch result {
+                    case .success(let token):
+                        completion(.success(.success(token)))
+                    case .failure(let error):
+                        switch error {
+                        case .apiClientError(let error):
+                            self.flow.process(error, completion: completion)
+                        case .timeout:
+                            completion(.failure(.pollingTimeoutExceeded))
+                        }
+                    }
+                    self.poll = nil
+                }
+            }
+        }
+    }
+    
+    // OOB authentication requests differ whether it's used as a primary factor, or a secondary factor.
+    // To simplify the code below, we separate this request logic into separate functions to work
+    // around differences in the response data.
+    func requestOOBCode(completion: @escaping (Result<OOBResponse, APIClientError>) -> Void) {
+        // Request where OOB is used as the primary factor
+        if let loginHint = loginHint {
+            requestOOBCode(loginHint: loginHint, completion: completion)
+        }
+        
+        // Request where OOB is used as the secondary factor
+        else if let mfaToken = mfaToken {
+            requestOOBCode(mfaToken: mfaToken, completion: completion)
```

Comment:
Should the requestOOBCode functions called by this function be private? It seems this function is taking on the responsibility of calling the correct one.

Reply:
I had left them `internal` to simplify testing, but in the end that wasn't necessary. So that's a good idea, marking them `private`.